### DISCUSSION TITLE
Changing Assets.find(EAssetType type).

### DISCUSSION THREAD

**Discussion by Jdance-Media** [SmartlyDressedGames/Unturned-3.x-Community#3689] [Ideas]
Currently if you try to get an array with ObjectAssets, you get a NotSupportedException, which seems pretty silly seeing as how you can use Assets.find<T>(List<T> results) to do basically the same thing. Proposing a small rewrite to the Assets.find(EAssetType type) method in order to prevent future confusion:
```csharp
    public static Asset[] find(EAssetType type)
    {
        switch (type)
        {
        case EAssetType.NONE:
            return null;
        case EAssetType.OBJECT:
            List<ObjectAsset> list = new List<ObjectAsset>();
            foreach (Asset asset in currentAssetMapping.assetList)
            {
                if (asset is ObjectAsset item)
                {
                    list.Add(item);
                }
            }
            return list.ToArray();
        default:
        {
            Asset[] array = new Asset[currentAssetMapping.legacyAssetsTable[type].Values.Count];
            int num = 0;
            {
                foreach (KeyValuePair<ushort, Asset> item in currentAssetMapping.legacyAssetsTable[type])
                {
                    array[num] = item.Value;
                    num++;
                }
                return array;
            }
        }
        }
    }
    ```

**Comment by SDGNelson** [comment]
Why are you using that method? Ideally any new code should use the version that takes a List instead. I have been working to get rid of usage of EAssetType.

**Comment by Jdance-Media** [comment]
My apologizes, wasn't aware of plans to make that obsolete. I found it easier to use that method since it returned an array that I was only going to use an iteration statement on, rather than having another line for a variable. Rather than my previous suggestion, perhaps there could be another method similar to the List one that just returns a list or an array for convenience?
